lib/internal/modules/cjs/loader.js

```diff
@@ -387,6 +386,43 @@ function resolveExports(nmPath, request, absoluteRequest) {
   return path.resolve(nmPath, request);
 }
 
+function resolveExportsTarget(pkgPath, target, subpath, basePath, mappingKey) {
+  if (typeof target === 'string') {
+    if (target.startsWith('./') &&
+        (subpath.length === 0 || target.endsWith('/'))) {
+      const resolvedTarget = new URL(target, pkgPath);
+      const pkgPathHref = pkgPath.href;
+      const resolvedTargetHref = resolvedTarget.href;
+      if (StringPrototype.startsWith(resolvedTargetHref, pkgPathHref) &&
+          StringPrototype.indexOf(resolvedTargetHref, '/node_modules/',
```

Comment:
this should be a check on `resolveTarget.pathname`

Reply:
Done.